src/inmanta/data/model.py

```diff
@@ -148,13 +148,44 @@ class CompileData(BaseModel):
     """
 
 
+class LinkObject(BaseModel):
+    """
+
+    :param href: a string whose value is a URI-reference pointing to the link’s target.
+
+    :param rel: a string indicating the link’s relation type. The string MUST be a valid link relation type.
+    :param describedby: a link to a description document (e.g. OpenAPI or JSON Schema) for the link target.
+    :param title: a string which serves as a label for the destination of a link
+        such that it can be used as a human-readable identifier (e.g., a menu entry).
+    :param type: a string indicating the media type of the link’s target.
+    :param hreflang: a string or an array of strings indicating the language(s) of the link’s target.
+        An array of strings indicates that the link’s target is available in multiple languages.
+        Each string MUST be a valid language tag [RFC5646].
+    :param meta: a meta object containing non-standard meta-information about the link.
+
+    """
+
+    href: str
+    rel: Optional[str] = None
+    describedby: Optional[str] = None
+    title: Optional[str] = None
+    type: Optional[str] = None
+    hreflang: Optional[str] = None
+    meta: Optional[dict[typing.Any, typing.Any]] = None
+
+
+JsonApiLinks: typing.TypeAlias = dict[str, Union[str, LinkObject, None]]
```

Comment:
```suggestion
type JsonApiLinks = dict[str, Union[str, LinkObject, None]]
```
New syntax

Reply:
And could you elaborate on this type? Why a dict, i.e. what would the keys be? Why the `str` in the `Union`?

Reply:
I used a dict instead of Mapping because I edit when joining the links of several compiles.
The str is there because we can simply pass a link there instead of building the `LinkObject` as per the jsonapi spec: https://jsonapi.org/format/#document-links

Reply:
> substituted_by

That is actually an interesting one. Not for now, but we may want to add that at some point.